File: ollama_api_example.js
const axios = require('axios');

// Configuração da API Ollama
const OLLAMA_URL = 'http://localhost:11434/api/generate';
const MODEL = 'qwen2:0.5b';

async function generateText(prompt, model = MODEL) {
    const payload = {
        model: model,
        prompt: prompt,
        stream: false
    };

    try {
        const response = await axios.post(OLLAMA_URL, payload, { timeout: 60000 });
        return response.data.response || '';
    } catch (error) {
        return `Erro: ${error.message}`;
    }
}

// Exemplos de uso
async function main() {
    // Exemplo 1: Conversa simples
    console.log('=== Exemplo 1: Conversa simples ===');
    const resposta1 = await generateText('Olá, como você está?');
    console.log(`Resposta: ${resposta1}\n`);

    // Exemplo 2: Tradução
    console.log('=== Exemplo 2: Tradução ===');
    const resposta2 = await generateText('Traduza para inglês: "Bom dia, mundo!"');
    console.log(`Resposta: ${resposta2}\n`);

    // Exemplo 3: Resumo
    console.log('=== Exemplo 3: Resumo ===');
    const texto = 'A inteligência artificial está transformando indústrias ao automatizar tarefas e analisar grandes volumes de dados.';
    const resposta3 = await generateText(`Resuma este texto em uma frase: ${texto}`);
    console.log(`Resposta: ${resposta3}\n`);

    // Exemplo 4: Código
    console.log('=== Exemplo 4: Geração de código ===');
    const resposta4 = await generateText('Escreva uma função Python que calcula o fatorial de um número.');
    console.log(`Resposta: ${resposta4}\n`);
}

main();

File: ollama_api_example.py
import requests
import json

# Configuração da API Ollama
OLLAMA_URL = "http://localhost:11434/api/generate"
MODEL = "qwen2:0.5b"

def generate_text(prompt, model=MODEL):
    """Gera texto usando o modelo Ollama via API."""
    payload = {
        "model": model,
        "prompt": prompt,
        "stream": False
    }
    
    try:
        response = requests.post(OLLAMA_URL, json=payload, timeout=60)
        response.raise_for_status()
        result = response.json()
        return result.get("response", "")
    except requests.exceptions.RequestException as e:
        return f"Erro: {e}"

# Exemplos de uso
if __name__ == "__main__":
    # Exemplo 1: Conversa simples
    print("=== Exemplo 1: Conversa simples ===")
    resposta = generate_text("Olá, como você está?")
    print(f"Resposta: {resposta}\n")

    # Exemplo 2: Tradução
    print("=== Exemplo 2: Tradução ===")
    resposta = generate_text("Traduza para inglês: 'Bom dia, mundo!'")
    print(f"Resposta: {resposta}\n")

    # Exemplo 3: Resumo
    print("=== Exemplo 3: Resumo ===")
    texto = "A inteligência artificial está transformando indústrias ao automatizar tarefas e analisar grandes volumes de dados."
    resposta = generate_text(f"Resuma este texto em uma frase: {texto}")
    print(f"Resposta: {resposta}\n")

    # Exemplo 4: Código
    print("=== Exemplo 4: Geração de código ===")
    resposta = generate_text("Escreva uma função Python que calcula o fatorial de um número.")
    print(f"Resposta: {resposta}\n")